### PR TITLE
Add support for custom comparators and reporters

### PR DESCRIPTION
**Pull request by juja0**

### CODE REVIEW

**Comment by jackmatt2** [comment]
@juja0 can you squash all the commits into a single one?

File: README.md [hunk 1]
```diff
@@ -317,6 +317,69 @@ public class HibernateSnapshotSerializer implements SnapshotSerializer {
...
+
+Comparators follow the same resolution order as Serializers
+1. method 
+2. class
+3. global
+
+### Example: JsonObjectComparator
+The default comparator may be too strict for certain types of data.
+For example, when comparing json objects, formatting of the json string or the order of fields 
+may not be of much importance during comparison. A custom serializer can help in such cases.
```
Comment (jackmatt2) [inline README.md:335]: a custom ~~serializer~~ comparator

**Comment by jackmatt2** [comment]
Did you end up trying`relocate`? Did it cause issues with the IDE being able to recognise the exceptions?

**Comment by juja0** [comment]
Hey, yes I'll squash the commits. Isn't that something that shows up as an option during merge though ? I don't remember seeing an option while raising the PR. Shall I squash during merge or do you want me to do it right now ? 

I did try relocating. It didn't work. Quoting my comment from the previous MR just in case you missed it.

> @jackmatt2
> Soft reset did not work for some reason. Tried a hard reset followed by a force push and the PR closed automatically so  I've raised a new one.
> #48
> Also intellij couldn't show proper diff after relocation so I've skipped it.

File: README.md (hunk 1 above)
Comment (juja0) [reply README.md:335]: Ah. Will change it.

**Comment by jackmatt2** [comment]
Ah yeah - didn't know Github supported automatic squash - that's nice.  Thanks for your help with these features - they offer some good enhancements.

**Pull request merged**